src/ejabberd_c2s.erl

```diff
@@ -376,13 +385,13 @@ maybe_compress_feature(SockMod, #state{zlib = {ZLib, _}}) ->
         _ -> []
     end.
 
-maybe_sasl_mechanisms(Server) ->
+maybe_sasl_mechanisms(#state{server = Server} = S) ->
```

Comment:
`State`?

Reply:
i like shortness, don't think that someone will get lost in 5 lines of code.